pkg/ibc/ibc.go

```diff
@@ -50,15 +65,94 @@ func GetClientsInfo(ibc *relayer.IBCdata, rpcs *map[string]config.RPC) (ClientsI
 
 	ctx := context.Background()
 
-	clientsInfo.ChainAClientExpiration, clientsInfo.ChainAClientInfo, err = relayer.QueryClientExpiration(ctx, chainA, chainB)
+	clientsInfo.ChainAClientExpiration, clientsInfo.ChainAClientInfo, err = relayer.QueryClientExpiration(
+		ctx,
+		chainA,
+		chainB,
+	)
 	if err != nil {
 		return ClientsInfo{}, fmt.Errorf("Error: %w path %v <-> %v", err, cdA, cdB)
 	}
 
-	clientsInfo.ChainBClientExpiration, clientsInfo.ChainBClientInfo, err = relayer.QueryClientExpiration(ctx, chainB, chainA)
+	clientsInfo.ChainBClientExpiration, clientsInfo.ChainBClientInfo, err = relayer.QueryClientExpiration(
+		ctx,
+		chainB,
+		chainA,
+	)
 	if err != nil {
 		return ClientsInfo{}, fmt.Errorf("Error: %w path %v <-> %v", err, cdB, cdA)
 	}
 
 	return clientsInfo, nil
 }
+
+func GetChannelInfo(ibc *relayer.IBCdata, rpcs *map[string]config.RPC) (ChannelsInfo, error) {
+	ctx := context.Background()
+	channelInfo := ChannelsInfo{}
+
+	if (*rpcs)[ibc.Chain1.ChainName].ChainID == "" || (*rpcs)[ibc.Chain2.ChainName].ChainID == "" {
+		return ChannelsInfo{}, fmt.Errorf(
+			"Error: RPC data is missing, cannot retrieve channel data",
+		)
+	}
+
+	cdA := chain.Info{
+		ChainID:  (*rpcs)[ibc.Chain1.ChainName].ChainID,
+		RPCAddr:  (*rpcs)[ibc.Chain1.ChainName].URL,
+		ClientID: ibc.Chain1.ClientID,
+	}
+
+	chainA, err := chain.PrepChain(cdA)
+	if err != nil {
+		return ChannelsInfo{}, fmt.Errorf("Error: %w for %v", err, cdA)
+	}
+
+	cdB := chain.Info{
+		ChainID:  (*rpcs)[ibc.Chain2.ChainName].ChainID,
+		RPCAddr:  (*rpcs)[ibc.Chain2.ChainName].URL,
+		ClientID: ibc.Chain2.ClientID,
+	}
+
+	chainB, err := chain.PrepChain(cdB)
+	if err != nil {
+		return ChannelsInfo{}, fmt.Errorf("Error: %w for %v", err, cdB)
+	}
+
+	for _, c := range ibc.Channels {
+		var order chantypes.Order
+
+		var channel Channel
+
+		channel.Name = c.Chain1.ChannelID
+
+		switch c.Ordering {
+		case "none":
+			order = chantypes.NONE
+		case "unordered":
+			order = chantypes.UNORDERED
+		case "ordered":
+			order = chantypes.ORDERED
+		}
+
+		ch := chantypes.IdentifiedChannel{
+			State:    3,
```

Comment:
Because it correlates to "STATE_OPEN".
This is because there's no good way of getting dynamically that certain object without making the request too complicated.

We can also assume that the channel status in this case, IMO.